blockchain/chainio.go

```diff
@@ -1331,40 +1346,31 @@ func dbFetchBlockByNode(dbTx database.Tx, node *blockNode) (*btcutil.Block, erro
 	return block, nil
 }
 
-// dbStoreBlock stores the provided block in the database. The block header is
-// written to the block index bucket and full block data is written to ffldb.
-func dbStoreBlockHeader(dbTx database.Tx, blockHeader *wire.BlockHeader, height uint32) error {
-	// Serialize block data to be stored. This is just the serialized header.
+// dbStoreBlockNode stores the block header and validation status to the block
+// index bucket. This overwrites the current entry if there exists one.
+func dbStoreBlockNode(dbTx database.Tx, node *blockNode) error {
+	// Serialize block data to be stored.
 	w := bytes.NewBuffer(make([]byte, 0, blockHdrSize))
```

Comment:
This should be `blockHdrSize+1` to account for the status byte as well.  Otherwise, two allocations will have to be done because it will serialize the header and then need more space which will force a realloc.